src/java/org/apache/cassandra/service/opstate/KeyspaceTableOpStateCache.java

```diff
@@ -63,6 +69,25 @@ public Optional<Instant> getMinimumTsOfAllEntries()
         if (tableEntries.isEmpty())
             return Optional.empty();
 
+        // Remove potentially deleted keyspaces and tables from entries
+        Set<KeyspaceTableKey> invalidEntries =
+            Sets.difference(tableEntries.keySet(), Sets.intersection(tableEntries.keySet(), getValidKeyspaceTableEntries()));
+        invalidEntries.forEach(tableEntries::remove);
+
         return Optional.of(Collections.min(tableEntries.values()));
```

Comment:
Is it possible we have a couple successful cleanups in the cache but we haven't been able to read from disk yet? In that case we would return the min of what's in the cache, which could be later than some other table that's logged on disk and avoid triggering a cleanup even if it might have been needed?

Reply:
Yes this is correct, but the assumption was that this behavior was acceptable. You would indeed miss one cleanup, but you would probably pick up on it at a later time (next time we successfully read from the file or next time a bootstrap is triggered).

Reply:
gotcha, if we've decided that's fine I'm cool with it, just wanted to make sure